Userland/Demos/CatDog/CatDog.cpp

```diff
@@ -10,174 +10,188 @@
 #include <LibGUI/Painter.h>
 #include <LibGUI/Window.h>
 
+namespace {
+
+enum State : uint16_t {
```

Comment:
Not the biggest fan of doing everything with a "raw" enum, you could split this up into `Direction`, `State` and `Frame` or something, although I understand that would complicate the image picking step.

Reply:
Since they're all animation states, I do think this is the way to go. Separating enums would bring all sorts of logic back etc. This is really the key to make animation frame selection simple and clean up that huge if/else/if tree that was there.
So I'd like to keep this.